packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -721,8 +721,9 @@ class FlutterPlugin implements Plugin<Project> {
                 return
             }
             // Flutter module included as a subproject in add to app.
-            Project appProject = project.rootProject.findProject(':app')
-            assert appProject != null
+            String mainAppModuleName = project.rootProject.hasProperty('app_module_name') ? project.rootProject.property('app_module_name')  : "app"
+            Project appProject = project.rootProject.findProject(":${mainAppModuleName}")
+            assert appProject != null : "Project with name :${mainAppModuleName} isn't existing, custom app project name you can define in gradle.properties app_module_name"
```

Comment:
What about naming this `flutter_host_app_project_name`? In the assert, this needs to be: `org.gradle.project.flutter_host_app_project_name`. e.g.

```
Project :${mainAppProjectName} doesn't exist. To custom the host app project name, set `org.gradle.project.flutter_host_app_project_name=<project-name>` in gradle.properties.
```

Reply:
updated